app/models/cart.rb

```diff
@@ -1,6 +1,8 @@
 class Cart < ApplicationRecord
   has_many :line_items, dependent: :destroy
-
+  has_many :products, through: :line_items
+  scope :enabled_products, -> { Product.where enabled: true }
```

Comment:
let's create this scope in product instead. 
Will this give enabled products of the cart or all the enabled products?

Reply:
Created that scope in product
Yes this scope would be giving all enabled_products and not just the enabled_products in the cart

Reply:
Fixed the issue